### PR TITLE
[DeadCode] Add RemovePhpVersionIdCheckRector

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/6448

### CODE REVIEW

**Comment by TomasVotruba** [comment]
This rule should be configurable by input PHP Version.
E.g. on PHP 7.4 and lower the test should be untouched

**Comment by samsonasik** [comment]
Ok, I will try

File: rules/DeadCode/Rector/ConstFetch/RemovePhpVersionIdCheckRector.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    /**
+     * @var string
+     */
+    public const PHP_VERSION_CONSTRAINT = '8.0';
+
+    /**
+     * @var array<string, string>
+     */
+    private const EXAMPLE_CONFIGURATION = [
+        self::PHP_VERSION_CONSTRAINT => '8.0',
```
Comment (staabm) [inline rules/DeadCode/Rector/ConstFetch/RemovePhpVersionIdCheckRector.php]: maybe as a second iteration: could this php version be read from `composer.json` as a config option?

Comment (samsonasik) [reply rules/DeadCode/Rector/ConstFetch/RemovePhpVersionIdCheckRector.php]: that was my thought https://github.com/rectorphp/rector/issues/6448#issuecomment-860648780 , @TomasVotruba need it to be configurable. I think that can be a fallback.

**Comment by samsonasik** [comment]
@TomasVotruba I think for make configurable, we need to more explicit config, currently, the config I made is like the following:

```php
    $services->set(RemovePhpVersionIdCheckRector::class)
        ->call('configure', [[
            RemovePhpVersionIdCheckRector::PHP_VERSION_CONSTRAINT => PhpVersion:: PHP_80,
        ]]);
```

which is ambiguous which greater/smaller to compare. I think we need to do something like:

```php
    $services->set(RemovePhpVersionIdCheckRector::class)
        ->call('configure', [[
            RemovePhpVersionIdCheckRector::PHP_VERSION_CONSTRAINT => PhpVersion::PHP_80,
            RemovePhpVersionIdCheckRector::KIND => 'smaller', // can be 'smallerOrEqual', 'greater', 'greaterOrEqual',
            RemovePhpVersionIdCheckRector::POSITION => 'left', // can be 'right'
        ]]);
```

which define which version to be check, what binary op, and where the position, so with above config, it make the following code:

```php
if (PHP_VERSION_ID < 80000) { // PHP_VERSION_ID in the 'left' with kind => 'smaller'
    echo 'to be deleted';
}
echo 'to be kept';
```

to be:

```php
echo 'to be kept';
```

what do you think?

**Comment by TomasVotruba** [comment]
> which define which version to be check, what binary op, and where the position, so with above config, it make the following code

I think the all the compares that return `true` should be included. Regardless left/right/smaller/greater.

**Comment by TomasVotruba** [comment]
This looks like easy merge. What needs to be done here?

**Comment by samsonasik** [comment]
The configurable make check ambiguous, eg in FixturePhp74 test, the assumption is happen as we are at php 8. 

I am thinking that we should rely to php version provider which check on current or composer.json only, what do you think?

**Comment by samsonasik** [comment]
I removed php 7.4 tests to avoid confusion as tested in php 8.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**